### ISSUE TITLE
Question about how optimus-manager works with lightdm

### ISSUE THREAD

**Issue by SuperBaguette** [Askannz/optimus-manager#265]
Hi ! I have a question regarding optimus-manager. It works fine on my laptop, so first of all : thank you very much for developping this handy software !

My situation is the following : I have a recent laptop with a Turing NVidia GPU and an 8th generation intel CPU. As a result, I can use the full power-saving capability of this configuration and I therefore I'm always running the hybrid mode of optimus-manager. 

I thought that it would be interesting to try and reproduce this mode without using optimus-manager, but using X11 config files instead. I therefore copied the config file generated by optimus-manager in /etc/X11/xorg.conf.d and disabled optimus-manager for now. 

When I start my graphical environment using startx, no problem : I can see both the intel and nvidia providers using xrandr --listproviders. However, when I enable lightdm, I only see the intel provider using the same command. I tried to copy some options generated by optimus-manager in /etc/lightdm/lightdm.conf.d (the line type=local) but it's not sufficient. 

In the code of optimus-manager, I did not find how it makes it all work under lightdm.
Investigating further using Xorg logs, it looks like when I'm using the Xorg config file alone and lightdm, the nvidia driver gets unloaded by xorg. However, when using startx and still the same Xorg configuration, the module stays loaded and therefore, everything works fine just as when I'm using optimus-manager. 

To sum up, could you please explain how optimus-manager makes everything work with lightdm when using the hybrid mode ?

Thank you ! If any additional information is needed, I'll happily provide it.
Have a good day

**Comment by Askannz** [comment]
Hi,

I wish I could help you, but optimus-manager doesn't do anything special for LightDM, besides hooking it to `/sbin/prime-offload`(which shouldn't be required for hybrid mode).

As a matter of fact, I just tried the same steps as you (copying the Xorg config from optimus-manager in hybrid mode, uninstalling it, enabling LightDM), and it worked. So I really don't see where the problem could be. Maybe the documentation of Nvidia can be of some help: http://download.nvidia.com/XFree86/Linux-x86_64/440.82/README/primerenderoffload.html

**Comment by SuperBaguette** [comment]
Hi @Askannz, 

Thanks a lot for looking into this, knowing that it works on your side is already helpful. I'll keep investigating, maybe something I installed in the past is interferring. In case I end up finding something of interest, I'll come back and post what was actually happening. 

Cheers, 
Have a good day
